cpp/src/prims/property_op_utils.cuh

```diff
@@ -136,6 +137,36 @@ struct intersection_op_result_type<
                                            raft::device_span<vertex_t const>>::type;
 };
 
+template <typename T>
+__device__ std::enable_if_t<std::is_arithmetic<T>::value, void> atomic_accumulate_impl(
+  thrust::detail::any_assign& /* dereferencing thrust::discard_iterator results in this type */ lhs,
+  T const& rhs)
+{
+  // no-op
+}
+
+template <typename T>
+__device__ std::enable_if_t<std::is_arithmetic<T>::value, void> atomic_accumulate_impl(T& lhs,
+                                                                                       T const& rhs)
+{
+  atomicAdd(&lhs, rhs);
+}
+
+template <typename Iterator, typename TupleType, size_t I, size_t N>
+struct atomic_accumulate_thrust_tuple_impl {
+  __device__ constexpr void compute(Iterator iter, TupleType const& value) const
+  {
+    atomic_accumulate_impl(thrust::raw_reference_cast(thrust::get<I>(*iter)),
+                           thrust::get<I>(value));
+    atomic_accumulate_thrust_tuple_impl<Iterator, TupleType, I + 1, N>().compute(iter, value);
+  }
+};
+
+template <typename Iterator, typename TupleType, size_t I>
+struct atomic_accumulate_thrust_tuple_impl<Iterator, TupleType, I, I> {
+  __device__ constexpr void compute(Iterator iter, TupleType const& value) const {}
+};
+
```

Comment:
Not related to to this PR: If the tuple has too many element (not sure if it can be the case), would it be a bit inefficient in terms of number of function instantiates it would generate?

Reply:
As we discussed offline, this will end up with N atomicAdd instructions. So, no in terms of execution time, but yes, this can increase compile time.